File: src/main/java/org/example/core/pages/games/LuckyFish.java
package org.example.core.pages.games;

import org.example.core.pages.ActionController;
import org.openqa.selenium.Point;

public class LuckyFish {

    private final Point play_btn = new Point(959, 750);
    private final Point bet_btn = new Point(957, 870);
    private final Point autobet_btn = new Point(800, 730);
    private final Point autocashout_btn = new Point(900, 730);
    private final Point game_info = new Point(1700, 185);
    private final Point sound_on = new Point(1740, 185);
    private final Point blockchain_info = new Point(1780, 290);
    private final Point game_history = new Point(1780, 230);

    private final Point minus_coef = new Point(1030, 730);
    private final Point plus_coef = new Point(1120, 730);
    private final Point input_field_coef = new Point(1070, 730);

    private final Point minus_bet = new Point(815, 780);
    private final Point plus_bet = new Point(970, 780);
    private final Point input_field_bet = new Point(900, 780);

    private final Point one = new Point(810, 820);
    private final Point two = new Point(870, 820);
    private final Point five = new Point(920, 820);
    private final Point ten = new Point(980, 820);

    private final Point balance = new Point(960, 940);

    private final Point reload_btn = new Point(963, 735);

    public void startGame() {
        ActionController.clickOn(play_btn);
    }

    public void turn_music() {
        ActionController.clickOn(sound_on);
    }

    public void show_game_history() {
        ActionController.clickOn(game_history);
    }

    public void show_blockchain_info() {
        ActionController.clickOn(blockchain_info);
    }

    public void show_gameinfo() {
        ActionController.clickOn(game_info);
    }

    public void bet() {
        ActionController.clickOn(bet_btn);
    }

    public void enable_autocashout() {
        ActionController.clickOn(autocashout_btn);
    }

    public void enable_autobet() {
        ActionController.clickOn(autobet_btn);
    }

    public void change_coef(boolean up, int howMany) {
        if (up) {
            for (int i = 0; i < howMany; i++) {
                ActionController.clickOn(plus_coef);
            }
        } else {
            for (int i = 0; i < howMany; i++) {
                ActionController.clickOn(minus_coef);
            }
        }
    }
}

File: src/main/java/org/example/core/pages/SC games/LuckyFish.kt
package org.example.core.pages.games;

import org.example.core.pages.ActionController;
import org.openqa.selenium.Point;

public class LuckyFish {

   private var isSound = false

   private val play_btn: Point = Point(959, 750);
   private val bet_btn: Point = Point(957, 870);
   private val autobet_btn: Point = Point(800, 730);
   private val autocashout_btn: Point = Point(900, 730);
   private val game_info: Point = Point(1700, 185);
   private val sound_on: Point = Point(1740, 185);
   private val blockchain_info: Point = Point(1780, 290);
   private val game_history: Point = Point(1780, 230);

   private val minus_coef: Point = Point(1030, 730);
   private val plus_coef: Point = Point(1120, 730);
   private val input_field_coef: Point = Point(1070, 730);

   private val minus_bet: Point = Point(815, 780);
   private val plus_bet: Point = Point(970, 780);
   private val input_field_bet: Point = Point(900, 780);

   private val one: Point = Point(810, 820);
   private val two: Point = Point(870, 820);
   private val five: Point = Point(920, 820);
   private val ten: Point = Point(980, 820);

   private val balance: Point = Point(960, 940);

   private val reload_btn: Point = Point(963, 735);

    public fun startGame() {
        ActionController.clickOn(play_btn);
    }

    public fun turn_music() {
        ActionController.clickOn(sound_on);
    }

    public fun show_game_history() {
        ActionController.clickOn(game_history);
    }

    public fun show_blockchain_info() {
        ActionController.clickOn(blockchain_info);
    }

    public fun show_gameinfo() {
        ActionController.clickOn(game_info);
    }

    public fun bet() {
        ActionController.clickOn(bet_btn);
    }

    public fun enable_autocashout() {
        ActionController.clickOn(autocashout_btn);
    }

    public fun enable_autobet() {
        ActionController.clickOn(autobet_btn);
    }

    public fun change_coef(boolean up, int howMany) {
        if (up) {
            for (int i = 0; i < howMany; i++) {
                ActionController.clickOn(plus_coef);
            }
        } else {
            for (int i = 0; i < howMany; i++) {
                ActionController.clickOn(minus_coef);
            }
        }
    }


}
